app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -46,6 +49,27 @@ public static LALogicLinesFragment newInstance(Bundle params, Context context) {
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         this.channelMode = params.getInt("channelMode");
+        switch (channelMode) {
+            case 1:
+                channelNames.add(params.getString("inputChannel1"));
+                break;
+            case 2:
+                channelNames.add(params.getString("inputChannel1"));
+                channelNames.add(params.getString("inputChannel2"));
+                break;
+            case 3:
+                channelNames.add(params.getString("inputChannel1"));
+                channelNames.add(params.getString("inputChannel2"));
+                channelNames.add(params.getString("inputChannel3"));
+                break;
+            case 4:
+                channelNames.add(params.getString("inputChannel1"));
+                channelNames.add(params.getString("inputChannel2"));
+                channelNames.add(params.getString("inputChannel3"));
+                channelNames.add(params.getString("inputChannel4"));
+                break;
+            default:
```

Comment:
Yeah, its simplified but that doesn't solve the purpose. I need to add channels in order to populate in the graph in same order.
I can use your way and reverse the list, but that would make code unnecessary difficult to understand for some new dev + reversing the list has its overhead too.